src/libsmac/src/parser.rs

```diff
@@ -4,7 +4,7 @@ use std::str::from_utf8;
 
 // Identifier
 // IDENT := /[a-zA-Z_]\w*/g
-named!(ident<&str, &str>, re_match!(r"[a-zA-Z_]\w*"));
+named!(ident<&str, &str>, re_match!(r"[a-zA-Z_]\w*[\?\!]*"));
```

Comment:
That would probably be better, yes. Forgot whether we discussed this. `[a-zA-Z_]\w*(?:\w|[\?\!])*` is perfect. :ok_hand: